guides/source/caching_with_rails.md

```diff
@@ -512,12 +512,23 @@ class ProductsController < ApplicationController
 end
 ```
 
-### A note on weak ETags
+### Notes on strong and weak ETags
+ETags generated by Rails are weak by default. Weak ETags allow semantically
+equivalent responses to have the same ETags, even if their bodies do not match
+exactly. This is useful when we don't want the page to be regenerated for minor
+changes in response body.
 
-Etags generated by Rails are weak by default. Weak etags allow semantically equivalent responses to have the same etags, even if their bodies do not match exactly. This is useful when we don't want the page to be regenerated for minor changes in response body. If you absolutely need to generate a strong etag, it can be assigned to the header directly.
 
 ```ruby
-  response.add_header "ETag", Digest::MD5.hexdigest(response.body)
+  response.weak_etag = response.body # => W/"hex string" → weak ETag
+```
+
+Strong ETag implies that response should be exactly same and byte by byte identical.
+Useful when doing Range requests within a large video or PDF file. Some CDNs
+support only strong ETags, like Akamai.
+
+```ruby
+  response.strong_etag = response.body # => "hex string" with no leading W/ → strong ETag
```

Comment:
Let's put a real example instead of `hex string` for example - `618bbc92e2d35ea1945008b42799b0e7`